locale/merge_csl_locales

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python3
+
+import json
+import os
+import sys
+import urllib.request
+import shutil
+import xml.etree.ElementTree as ET
+import unicodedata
+
+LOCALES_GIT_REF = "bd8d2dbc85713b192d426fb02749475df30f0d2c"
+
+
+def main():
+    use_cache = True
+
+    script_dir = os.path.dirname(os.path.realpath(__file__))
+    if len(sys.argv) < 2 or not os.path.isdir(sys.argv[1]):
+        sys.stderr.write(
+            "Usage: {0} path/to/zotero/source\n".format(os.path.basename(sys.argv[0]))
+        )
+        return 1
+
+    source_dir = sys.argv[1]
+    schema_dir = os.path.join(source_dir, "resource", "schema")
+
+    csl_locale_base = "https://raw.githubusercontent.com/citation-style-language/locales/{ref}/locales-{lang}.xml"
+
+    # Codes for the language packs that we want to grab
+    language_packs = [
+        "af-ZA",
+        "ar",
+        "bg-BG",
+        "ca-AD",
+        "cs-CZ",
+        "cy-GB",
+        "da-DK",
+        "de-AT",
+        "de-CH",
+        "de-DE",
+        "el-GR",
+        "en-GB",
+        "en-US",
+        "es-CL",
+        "es-ES",
+        "es-MX",
+        "et-EE",
+        "eu",
+        "fa-IR",
+        "fi-FI",
+        "fr-CA",
+        "fr-FR",
```

Comment:
I didn't check very closely, but I'd be surprised if the different regional variants are actually any different. (Even `zh-CN` and `zh-TW` appear to be the same.) I suspect we can just pick one for each (e.g., just `es-ES`) and save it with the primary language code only (`es`), which will also make choosing a locale easier (avoiding the need for `Zotero.Utilities.Internal.resolveLocale()`).

Reply:
`pt-PT` and `pt-BR` are (different punctuation and gender agreement rules, it looks like), as are some of the Spanishes. I suppose we could make the regex a bit more accepting when it comes to punctuation and it *should* even out the differences as long as we choose the right variant for each one (e.g. we have to base our Portuguese locale on `pt-BR`). But I think it might be worth just using `resolveLocale`, just in case.